File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack=[]
    flag=True
    
    for i in range(len(s)):
        if s[i]=='(':
            stack.append('(')
        else:
            if not stack:
                flag=False
                break
            stack.pop()
            
    if not stack and flag==True:
        return True
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        for(char ch:s.toCharArray()){
            if(ch=='('){
                stack.add(ch);
            }
            else if(!stack.isEmpty()){
                stack.remove();
            }
            else{
                return false;
            }
        }
        return stack.isEmpty();
    }
}